cmd/cluster-init/generate_job.go

```diff
@@ -0,0 +1,158 @@
+				LabelRole: Infra,
+			},
+		},
+		Interval: "12h",
+	}
+}
+
+func generatePostsubmit(clusterName string) prowconfig.Postsubmit {
+	return prowconfig.Postsubmit{
+		JobBase: prowconfig.JobBase{
+			Name:       RepoMetadata().JobName(jobconfig.PostsubmitPrefix, clusterName+"-apply"),
+			Agent:      string(prowapi.KubernetesAgent),
+			Cluster:    string(api.ClusterAPPCI),
+			SourcePath: "",
+			Spec: &v1.PodSpec{
+				Volumes: []v1.Volume{generateSecretVolume(clusterName)},
+				Containers: []v1.Container{
+					generateContainer(LatestImage, clusterName, []string{"--confirm=true"}, []v1.VolumeMount{})},
+				ServiceAccountName: ConfigUpdater,
+			},
+			UtilityConfig: prowconfig.UtilityConfig{
+				Decorate: utilpointer.BoolPtr(true),
+			},
+			MaxConcurrency: 1,
+			Labels: map[string]string{
+				LabelRole: Infra,
+			},
+		},
+		Brancher: prowconfig.Brancher{
+			Branches: []string{"master"},
+		},
+	}
+}
+
+func generatePresubmit(clusterName string) prowconfig.Presubmit {
+	return prowconfig.Presubmit{
+		JobBase: prowconfig.JobBase{
+			Name:       RepoMetadata().JobName(jobconfig.PresubmitPrefix, clusterName+"-dry"),
+			Agent:      string(prowapi.KubernetesAgent),
+			Cluster:    string(api.ClusterAPPCI),
+			SourcePath: "",
+			Spec: &v1.PodSpec{
+				Volumes: []v1.Volume{generateSecretVolume(clusterName),
+					{
+						Name: "tmp",
+						VolumeSource: v1.VolumeSource{
+							EmptyDir: &v1.EmptyDirVolumeSource{},
+						},
+					}},
+				Containers: []v1.Container{
+					generateContainer(LatestImage,
+						clusterName,
+						[]string{},
+						[]v1.VolumeMount{{Name: "tmp", MountPath: "/tmp"}})},
+				ServiceAccountName: ConfigUpdater,
+			},
+			UtilityConfig: prowconfig.UtilityConfig{Decorate: utilpointer.BoolPtr(true)},
+			Labels: map[string]string{
+				"pj-rehearse.openshift.io/can-be-rehearsed": "true",
+			},
+		},
+		AlwaysRun:    true,
+		Optional:     false,
+		Trigger:      prowconfig.DefaultTriggerFor(clusterName),
+		RerunCommand: prowconfig.DefaultRerunCommandFor(clusterName) + "-dry",
+		Brancher: prowconfig.Brancher{
+			Branches: []string{"master"},
+		},
+		Reporter: prowconfig.Reporter{
+			Context: fmt.Sprintf("ci/build-farm/%s-dry", clusterName),
+		},
+	}
+}
+
+func generateSecretVolume(clusterName string) v1.Volume {
+	return v1.Volume{
+		Name: "build-farm-credentials",
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName: "build-farm-credentials",
+				Items: []v1.KeyToPath{
+					{
+						Key:  serviceAccountKubeconfigPath(ConfigUpdater, clusterName),
+						Path: "kubeconfig",
+					},
+				},
+			},
+		},
+	}
+}
+
+func generateContainer(image, clusterName string, extraArgs []string, extraVolumeMounts []v1.VolumeMount) v1.Container {
+	return v1.Container{
+		Name:    "",
+		Image:   image,
+		Command: []string{"applyconfig"},
+		Args: append([]string{
+			fmt.Sprintf("--config-dir=clusters/build-clusters/%s", clusterName),
+			"--as=",
```

Comment:
Is this really what we want?

Reply:
https://github.com/openshift/ci-tools/pull/542
If I understand the PR correctly, `--as=` means that the flag is set with value "" so it wont use `--as system:admin` to impersonate as default.

Removing the line would mean the flag is not set and the default `--as system:admin` will have trouble.